### DISCUSSION TITLE
[auth] Add login parameter to createUserWithEmailAndPassword()

### DISCUSSION THREAD

**Discussion by jonbhanson** [firebase/flutterfire#5692] [Feature request]
Currently, when a user creates a new account in auth using email, the user is automatically logged in regardless of whether the email has been verified.  It seems most apps are designed to only allow login after the email is verified.  Currently, to implement this behavior it is necessary to immediately log the user out after the account is created.  This can create stutter if the app uses a `StreamBuilder` based on the the user stream.  For more detail, see [this stack overflow thread](https://stackoverflow.com/questions/57762736/firebase-auth-createuserwithemailandpassword-prevent-login-until-email-is).

Please add a `login` parameter to `createUserWithEmailAndPassword()` to allow the account to be created without logging in the account such as:

```dart
createUserWithEmailAndPassword(login: false);
```

This will save us from the current hacky solution.

**Comment by endgameapp** [comment]
This is something we really need guys. Its part of best practice authentication flows...

**Comment by sohodoll** [comment]
this is still relevant

**Comment by prggTheProgrammer** [comment]
I agree, but I think it's a bit hard for flutterfire to do this as flutterfire uses the underlying sdks. That means that I think you should create a feature request at the firebase sdk.
